core/src/main/scala/io/finch/ToService.scala

```diff
@@ -94,19 +97,25 @@ object ToService {
     def apply(
         endpoints: Endpoint[A] :: ET,
         includeDateHeader: Boolean,
-        includeServerHeader: Boolean): Service[Request, Response] = new Service[Request, Response] {
+        includeServerHeader: Boolean,
+        methodNotAllowed: Boolean): Service[Request, Response] = new Service[Request, Response] {
 
       private[this] val underlying = endpoints.head.handle(respond400OnErrors)
 
       def apply(req: Request): Future[Response] = underlying(Input.fromRequest(req)) match {
-        case EndpointResult.Matched(rem, out) if rem.route.isEmpty => out.map(oa => conformHttp(
-          oa.toResponse(trA, trE),
-          req.version,
-          includeDateHeader,
-          includeServerHeader
-        )).run
-
-        case _ => tsT(endpoints.tail, includeDateHeader, includeServerHeader)(req)
+        case EndpointResult.Matched(rem, out) if rem.route.isEmpty =>
+          if (rem.method == req.method) {
+            out.map(oa => conformHttp(
+              oa.toResponse(trA, trE),
+              req.version,
+              includeDateHeader,
+              includeServerHeader
+            )).run
+          } else {
+            tsT(endpoints.tail, includeDateHeader, includeServerHeader, methodNotAllowed = true)(req)
```

Comment:
Oh I see it now - this what makes it fall through to the next endpoint.

Reply:
Yep. I've updated a spec to reflect this case

Reply:
I like the idea of notifying the next endpoint that the match was found, but I think we might want to use a different flag for this (while keeping the `methodNotAllowed` unchanged). What do you think?

Reply:
Could you elaborate on this one, please? I don't get it

Reply:
Oh sorry. I think it's just me not understanding this change completely. At first, I thought `methodNotAllowed` is something that goes into `Bootstrap` so users can toggle it on/off (which I think we should do anyway). Then I figured you override it to signal the next endpoint that the match was found so I suggested using a different flag for this (turns out you already do that!). How about we wire in `Bootstrap.configured(enableMethodNotAllowed)`?

Also I'd probably vote to change `ToService.apply(methodNotAllowed)` to something like `matchIsFound` (or `endpointIsMatched`). What do you think?

Reply:
Ah, okay, got it.
But the match is actually not found because a method is incorrect.

Regarding configuration. I guess flag from the user should enable/disable the option instead of corrupting a behavior because it will turn all 404s into 405s if one will set it as `true` by default.

Reply:
Right. So ToService should have both: one from a user (whether or not to support 405), one from the previous endpoint (whether or not it was matched).

Reply:
How about `matchedWithoutMethod` flag then (to indicate that very special case)?